lib/preprocess.js

```diff
@@ -0,0 +1,42 @@
+import assert from 'node:assert'
+import { ethers } from 'ethers'
+
+export const preprocess = async ({
+  rounds,
+  cid,
+  roundIndex,
+  web3Storage,
+  logger
+}) => {
+  const measurements = (await fetchMeasurements(web3Storage, cid))
+    .filter(measurement => {
+      try {
+        assertValidMeasurement(measurement)
+        return true
+      } catch (err) {
+        logger.error('Invalid measurement', err.message, measurement)
+        return false
+      }
+    })
+  logger.log(`Fetched ${measurements.length} valid measurements`)
+
+  if (!rounds[roundIndex]) {
+    rounds[roundIndex] = []
+  }
+  rounds[roundIndex].push(...measurements)
+}
+
+const fetchMeasurements = async (web3Storage, cid) => {
+  const res = await web3Storage.get(cid)
+  const files = await res.files()
+  const measurements = JSON.parse(await files[0].text())
+  return measurements
```

Comment:
Observability (after this PR is landed): I would consider reporting the number of measurements as a new telemetry point and visualising it in Grafana. I think both the "number of measurements per CID" and "number of measurements per round" are potentially useful metrics to watch.

Reply:
https://github.com/filecoin-station/spark-evaluate/issues/2